src/actions.js

```diff
@@ -96,7 +96,7 @@ export function callWaiter(name, { requestCreator, params }) {
     const waiterData = getWaiter(getState(), name);
 
     // Check if waiter is currently handling a request
-    if (waiterData.request && waiterData.isPending) {
+    if (waiterData.request && waiterData.request.then && waiterData.isPending) {
```

Comment:
Is `waiterData.request` the function that was given to the waiter as the `requestCreator` or is it the the request payload?

Reply:
It's the promise returned from the `requestCreator()`.